### DISCUSSION TITLE
Tasmota with Scripting too Large

### DISCUSSION THREAD

**Discussion by doceave** [arendst/Tasmota#17696] [Q&A] [answered]
I have secured a handful of these: QS-WiFi-D01-TRIAC; https://templates.blakadder.com/qs-wifi_D01_dimmer.html
Flashed with Tasmota using Tuya Convert and then updated to the latest version of Tasmota.

Issue now is that the device requires a Tasmota build with scripting. The available builds, when firmware update is attempted, result in an error "Not enough space".

I there someone who can point me to a minimalist Tasmota with Scripting? 
The builds available here are purportedly too bulky.

Thanks in advance.

**Comment by sfromis** [comment, marked as answer]
If you did not try upgrading via the `.gz` variant of the binary (you did not tell what you specifically tried), that's what you *must* use. Recommended is to not download first, but point directly at the http URL from the OTA site.

**Reply by doceave** [reply]
Thanks for your help. This does look to be one step closer to success however:
> OTA flashing proceeds, no errors, device restarts.... but no obvious scripting options seem to exist under the configuration page
> Information page also indicates that the firmware build is from 2022-12-16 

Is there something else obvious that I may be missing?

![Screenshot 2023-01-14 065245](https://user-images.githubusercontent.com/323727/212455681-dfac0757-19c2-40fd-87d5-e67f825fbfb9.jpg)

**Reply by barbudor** [reply]
There is no scripting in Tasmota lite
You need a tasmota binary with the scripting option
You can use Tasmota-scripting.bin.gz from here https://github.com/tasmota/install/tree/main/firmware/unofficial

**Reply by sfromis** [reply]
Beware that you cannot use the upgrade dialog pointing OTA directly at the URL from the Github site, as only http (not https) is supported for that. Hence, you need to download the file, and first upgrade to tasmota-minimal.bin.gz, and then to the downloaded scripting binary. That should work.

**Reply by barbudor** [reply]
As the current firmare is tasmota-lite, mstepping through minimal is probably not needed

**Reply by sfromis** [reply]
Testing an upgrade from lite to scripting, I observe minimal to also be needed for this step.

**Comment by doceave** [comment]
You have all been most gracious... Steps which worked:
> Tuya-Convert
> Upgrade to latest lite firmware OTA
> Step to minimal firmware OTA
> Download and flash tasmota-scripting.bin.gz
> Copy script to device as per: https://templates.blakadder.com/qs-wifi_D01_dimmer.html

Thanks again!

**Comment by doceave** [comment]
One caveat... the device name appears to be fixed. I.E. cannot be changed from the default "Tasmota" --- I will raise this in a new thread
